### ISSUE TITLE
ENOMEM error when building Strider-CD

### ISSUE THREAD

**Issue by olalonde** [progrium/dokku#1243]
I'm trying to install Strider-CD (https://github.com/Strider-CD/strider) on Dokku and getting this error at initial `git push dokku`:

```bash
       70867 info install ws@0.5.0
       70868 verbose unbuild node_modules/socket.io-client/node_modules/engine.io-client/node_modules/engine.io-parser/node_modules/has-binary/node_modules/isarray
       70869 http 304 https://registry.npmjs.org/sigmund
       70870 silly get cb [ 304,
       70870 silly get   { date: 'Tue, 23 Jun 2015 07:34:00 GMT',
       70870 silly get     via: '1.1 varnish',
       70870 silly get     'cache-control': 'max-age=60',
       70870 silly get     etag: '"1TUX00X8YEKNFED6GGWS214S3"',
       70870 silly get     age: '42',
       70870 silly get     connection: 'keep-alive',
       70870 silly get     'x-served-by': 'cache-hkg6827-HKG',
       70870 silly get     'x-cache': 'HIT',
       70870 silly get     'x-cache-hits': '2',
       70870 silly get     'x-timer': 'S1435044840.012203,VS0,VE0',
       70870 silly get     vary: 'Accept' } ]
       70871 verbose etag https://registry.npmjs.org/sigmund from cache
       70872 verbose get saving sigmund to /app/.npm/registry.npmjs.org/sigmund/.cache.json
       70873 verbose unsafe-perm in lifecycle true
       70874 verbose stack Error: spawn ENOMEM
       70874 verbose stack     at exports._errnoException (util.js:746:11)
       70874 verbose stack     at ChildProcess.spawn (child_process.js:1162:11)
       70874 verbose stack     at exports.spawn (child_process.js:995:9)
       70874 verbose stack     at spawn (/tmp/build/vendor/node/lib/node_modules/npm/lib/utils/spawn.js:7:13)
       70874 verbose stack     at runCmd_ (/tmp/build/vendor/node/lib/node_modules/npm/lib/utils/lifecycle.js:207:14)
       70874 verbose stack     at runCmd (/tmp/build/vendor/node/lib/node_modules/npm/lib/utils/lifecycle.js:172:5)
       70874 verbose stack     at runPackageLifecycle (/tmp/build/vendor/node/lib/node_modules/npm/lib/utils/lifecycle.js:134:3)
       70874 verbose stack     at Array.<anonymous> (/tmp/build/vendor/node/lib/node_modules/npm/node_modules/slide/lib/bind-actor.js:15:8)
       70874 verbose stack     at LOOP (/tmp/build/vendor/node/lib/node_modules/npm/node_modules/slide/lib/chain.js:15:14)
       70874 verbose stack     at chain (/tmp/build/vendor/node/lib/node_modules/npm/node_modules/slide/lib/chain.js:20:5)
       70875 verbose cwd /tmp/build
       70876 error Linux 3.13.0-53-generic
       70877 error argv "/tmp/build/vendor/node/bin/node" "/tmp/build/vendor/node/bin/npm" "install" "--userconfig" "/tmp/build/.npmrc" "--production"
       70878 error node v0.12.5
       70879 error npm  v2.11.2
       70880 error code ENOMEM
       70881 error errno ENOMEM
       70882 error syscall spawn
       70883 error spawn ENOMEM
       70884 error If you need help, you may report this error at:
       70884 error     <https://github.com/npm/npm/issues>
       70885 verbose exit [ 1, true ]
```

I suppose my Node process is running out of memory and was wondering if there was a way to fix the problem.

**Comment by michaelshobbs** [comment]
There are no memory restrictions for dokku managed containers out of the box. Meaning your node process has access to all available system memory. 

Perhaps the VM/instance provisioned is too small to begin with?

**Comment by olalonde** [comment]
Yes, upgrading to 2GB memory fixed the problem, though I'm seeing a new error now:

```bash
ubuntu@d:~/strider$ git push dokku HEAD:refs/heads/master
Counting objects: 10156, done.
Delta compression using up to 2 threads.
Compressing objects: 100% (4351/4351), done.
Writing objects: 100% (10156/10156), 6.13 MiB | 0 bytes/s, done.
Total 10156 (delta 5532), reused 10156 (delta 5532)
-----> Cleaning up...
-----> Building strider from buildstep...
-----> Adding BUILD_ENV to build environment...
-----> Multipack app detected
       =====> Downloading Buildpack: https://github.com/mbuchetics/heroku-buildpack-nodejs-grunt
       =====> Detected Framework: Node.js

       PRO TIP: Avoid using semver ranges starting with '>' in engines.node
       See https://devcenter.heroku.com/articles/nodejs-support

-----> Requested node range: > 0.8.x
-----> Resolved node version: 0.12.5
-----> Downloading and installing node
-----> Restoring node_modules directory from cache
-----> Pruning cached dependencies not specified in package.json
       npm WARN package.json strider@1.6.6 No license field.
-----> Writing a custom .npmrc to circumvent npm bugs
-----> Exporting config vars to environment
-----> Installing dependencies
       npm WARN package.json strider@1.6.6 No license field.

       > strider@1.6.6 postinstall /tmp/build
       > bower install --allow-root && npm run build

       bower jquery-timeago#1.3.0  not-cached git://github.com/rmm5t/jquery-timeago.git#1.3.0
       bower jquery-timeago#1.3.0     resolve git://github.com/rmm5t/jquery-timeago.git#1.3.0


[.... some more logs ....]

       > strider@1.6.6 styles /tmp/build
       > lessc client/styles/strider.less > dist/styles/styles.css

-----> Caching node_modules directory for future builds
-----> Cleaning up node-gyp and npm artifacts
-----> Building runtime environment
       =====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-ruby.git
To dokku@localhost:strider
 ! [remote rejected] HEAD -> master (pre-receive hook declined)
error: failed to push some refs to 'dokku@localhost:strider'
```

**Comment by olalonde** [comment]
Opened a new issue.
